src/sagemaker_training/smdataparallel.py

```diff
@@ -164,6 +164,8 @@ def _get_mpirun_command(
             "RDMAV_FORK_SAFE=1",
             "-x",
             "LD_PRELOAD=%s" % inspect.getfile(gethostname),
+            "-x",
```

Comment:
EFA is enabled on only certain instance types. Please add a check and use extend to add this option.

Reply:
EFA is only enabled on `["ml.p3dn.24xlarge", "ml.p4d.24xlarge"]`. added the check in the commit now.